contribute/dev_contribution.rst

```diff
@@ -0,0 +1,110 @@
+.. Contribution documentation for OpenCMISS
+
+============
+Contributing
+============
+
+This document covers the process to follow for getting your changes into the :term:`prime repository`.  While there are many types of contribution, this section focuses on contributions made through GitHub and Git, or in other words assets that are managed using the version control system.  It is assumed that :doc:`Setup <dev_setup>` and :doc:`Building <dev_building>` have already been read and followed.
+
+.. contents::
+
+Overview
+========
+
+For any body of work intended for the :term:`prime repository` start with a GitHub issue.  The issue can be used to discuss the topic and clarify any problems related to it.  Once progress has been made towards addressing the issue a pull request is created that references the issue.
+
+Reviewers provide feedback on the changes by adding comments to the pull request or associated commits. The Jenkin's build/test procedure will run each time changes are pushed to the pull request's branch, and the results are displayed in the pull request view.
+
+Once all the changes and reviews are complete, one of the :term:`prime repository` owners will merge the pull request into the prime repository, onto the develop branch.
+
+Note that a bug is just a type of issue, and that resolving the bug should have both the implementation to fix the bug and a test that triggers the bug.
+
+:numref:`Figure %s <OpenCMISS-development-process>` gives a graphical overview of the developer contribution process.  For more detail see the text below.
+
+.. _OpenCMISS-development-process:
+.. figure:: images/OpenCMISSProcesses-DevelopmentProcess.png
+   :align: center
+   :alt: Developer contribution process
+
+   Developer contribution process
+
+GitHub Issue
+============
+
+If an issue does not exist for the required work (e.g. implementation of a feature, fixing of a bug), then create a new one.  The issue is the place to discuss the particulars related to the issue, discussions on determining the scope of the issue or clarification of any points that are unclear.
+
```

Comment:
Add: Create the issue into issue list for the respective component repository it relates to. [What if it relates to multiple components? perhaps suggest creating an issue in each and adding a comment linking to the other e.g. opencmiss/zinc#60, as documented in https://github.com/blog/967-github-secrets]